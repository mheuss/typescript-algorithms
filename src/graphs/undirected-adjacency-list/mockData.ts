export interface IStates {
  capital: string;
}

export interface IDistance {
  miles: number;
}
