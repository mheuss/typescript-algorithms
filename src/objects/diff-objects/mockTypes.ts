export interface IBankData {
  routing_number?: number;
  bank_name?: string;
  city: string;
  state: string;
  zip: string;
}

export type Banks = IBankData[];
