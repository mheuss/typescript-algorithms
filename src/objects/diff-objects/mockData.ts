import { Banks } from "./mockTypes";

export const originalFile: Banks = [
  {
    bank_name: "Tait",
    city: "Saint Louis",
    routing_number: 316489490,
    state: "Missouri",
    zip: "63126",
  },
  {
    bank_name: "Merle",
    city: "Denver",
    routing_number: 732919062,
    state: "Colorado",
    zip: "80930",
  },
  {
    bank_name: "Francis",
    city: "Worcester",
    routing_number: 202463187,
    state: "Massachusetts",
    zip: "01605",
  },
  {
    bank_name: "Kirsteni",
    city: "Seattle",
    routing_number: 279027748,
    state: "Washington",
    zip: "98104",
  },
  {
    bank_name: "Danielle",
    city: "Midland",
    routing_number: 151532905,
    state: "Texas",
    zip: "78703",
  },
  {
    bank_name: "Shelby",
    city: "Shreveport",
    routing_number: 149772497,
    state: "Louisiana",
    zip: "71130",
  },
  {
    bank_name: "Carolin",
    city: "Saint Paul",
    routing_number: 617706482,
    state: "Minnesota",
    zip: "55103",
  },
  {
    bank_name: "Bili",
    city: "Saint Paul",
    routing_number: 527362918,
    state: "Minnesota",
    zip: "55146",
  },
  {
    bank_name: "Karilynn",
    city: "Honolulu",
    routing_number: 224887802,
    state: "Hawaii",
    zip: "96850",
  },
  {
    bank_name: "Ward",
    city: "Chicago",
    routing_number: 331421157,
    state: "Illinois",
    zip: "61656",
  },
  {
    bank_name: "Jolynn",
    city: "Houston",
    routing_number: 920233367,
    state: "Texas",
    zip: "77010",
  },
  {
    bank_name: "Dorelia",
    city: "Memphis",
    routing_number: 547444708,
    state: "Tennessee",
    zip: "38131",
  },
  {
    bank_name: "Gwenni",
    city: "Wichita",
    routing_number: 701886244,
    state: "Kansas",
    zip: "67215",
  },
  {
    bank_name: "Jaime",
    city: "Indianapolis",
    routing_number: 760086826,
    state: "Indiana",
    zip: "46295",
  },
  {
    bank_name: "Phillipe",
    city: "Largo",
    routing_number: 145543598,
    state: "Florida",
    zip: "34643",
  },
  {
    bank_name: "Samson",
    city: "Jamaica",
    routing_number: 593655726,
    state: "New York",
    zip: "11436",
  },
  {
    bank_name: "Delmer",
    city: "Denver",
    routing_number: 423427452,
    state: "Colorado",
    zip: "80270",
  },
  {
    bank_name: "Walsh",
    city: "Oklahoma City",
    routing_number: 228880440,
    state: "Oklahoma",
    zip: "73135",
  },
  {
    bank_name: "Trula",
    city: "Bozeman",
    routing_number: 811244802,
    state: "Montana",
    zip: "59771",
  },
  {
    bank_name: "Viv",
    city: "Albuquerque",
    routing_number: 692682035,
    state: "New Mexico",
    zip: "87180",
  },
  {
    bank_name: "Dollie",
    city: "San Antonio",
    routing_number: 707249543,
    state: "Texas",
    zip: "78291",
  },
  {
    bank_name: "Bernardo",
    city: "Richmond",
    routing_number: 859971484,
    state: "Virginia",
    zip: "23293",
  },
  {
    bank_name: "Rosita",
    city: "Albany",
    routing_number: 675519439,
    state: "New York",
    zip: "12237",
  },
  {
    bank_name: "Erin",
    city: "Denver",
    routing_number: 498873111,
    state: "Colorado",
    zip: "80209",
  },
  {
    bank_name: "Melba",
    city: "Clearwater",
    routing_number: 535335163,
    state: "Florida",
    zip: "33758",
  },
  {
    bank_name: "Sandor",
    city: "New Orleans",
    routing_number: 916490031,
    state: "Louisiana",
    zip: "70183",
  },
  {
    bank_name: "Petronella",
    city: "Omaha",
    routing_number: 747464677,
    state: "Nebraska",
    zip: "68134",
  },
  {
    bank_name: "Eunice",
    city: "Lansing",
    routing_number: 407981014,
    state: "Michigan",
    zip: "48930",
  },
  {
    bank_name: "Mannie",
    city: "New York City",
    routing_number: 304952129,
    state: "New York",
    zip: "10184",
  },
  {
    bank_name: "Avery",
    city: "Tulsa",
    routing_number: 936501142,
    state: "Oklahoma",
    zip: "74126",
  },
  {
    bank_name: "Valene",
    city: "Fairbanks",
    routing_number: 173060444,
    state: "Alaska",
    zip: "99790",
  },
  {
    bank_name: "Janifer",
    city: "Lubbock",
    routing_number: 530484329,
    state: "Texas",
    zip: "79491",
  },
  {
    bank_name: "Griffin",
    city: "Sacramento",
    routing_number: 288115184,
    state: "California",
    zip: "94297",
  },
  {
    bank_name: "Lilly",
    city: "Kissimmee",
    routing_number: 842286875,
    state: "Florida",
    zip: "34745",
  },
  {
    bank_name: "Yoko",
    city: "Washington",
    routing_number: 184481379,
    state: "District of Columbia",
    zip: "20337",
  },
  {
    bank_name: "Cherlyn",
    city: "North Little Rock",
    routing_number: 514038385,
    state: "Arkansas",
    zip: "72118",
  },
  {
    bank_name: "Pen",
    city: "Los Angeles",
    routing_number: 937516362,
    state: "California",
    zip: "90101",
  },
  {
    bank_name: "Court",
    city: "Washington",
    routing_number: 426945562,
    state: "District of Columbia",
    zip: "20503",
  },
  {
    bank_name: "Marcy",
    city: "Norfolk",
    routing_number: 154955464,
    state: "Virginia",
    zip: "23520",
  },
  {
    bank_name: "Kizzie",
    city: "Austin",
    routing_number: 588297101,
    state: "Texas",
    zip: "78732",
  },
  {
    bank_name: "Marlane",
    city: "Peoria",
    routing_number: 312666186,
    state: "Illinois",
    zip: "61651",
  },
  {
    bank_name: "Parker",
    city: "Jackson",
    routing_number: 473229698,
    state: "Mississippi",
    zip: "39282",
  },
  {
    bank_name: "Kylila",
    city: "Birmingham",
    routing_number: 310231910,
    state: "Alabama",
    zip: "35295",
  },
  {
    bank_name: "Harmony",
    city: "Charlotte",
    routing_number: 270117236,
    state: "North Carolina",
    zip: "28210",
  },
  {
    bank_name: "Dehlia",
    city: "Lansing",
    routing_number: 713767044,
    state: "Michigan",
    zip: "48901",
  },
];

export const newFile: Banks = [
  {
    bank_name: "Tait",
    city: "Saint Louis",
    routing_number: 316489490,
    state: "Missouri",
    zip: "63126",
  },
  {
    bank_name: "Merle",
    city: "Colorado Springs",
    routing_number: 732919062,
    state: "Colorado",
    zip: "80930",
  },
  {
    bank_name: "Francis",
    city: "Worcester",
    routing_number: 202463187,
    state: "Massachusetts",
    zip: "01605",
  },
  {
    bank_name: "Kirsteni",
    city: "Seattle",
    routing_number: 279027748,
    state: "Washington",
    zip: "98104",
  },
  {
    bank_name: "Danielle",
    city: "Austin",
    routing_number: 151532905,
    state: "Texas",
    zip: "78703",
  },
  {
    bank_name: "Shelby",
    city: "Shreveport",
    routing_number: 149772497,
    state: "Louisiana",
    zip: "71130",
  },
  {
    bank_name: "Carolin",
    city: "Saint Paul",
    routing_number: 617706482,
    state: "Minnesota",
    zip: "55103",
  },
  {
    bank_name: "Bili",
    city: "Saint Paul",
    routing_number: 527362918,
    state: "Minnesota",
    zip: "55146",
  },
  {
    bank_name: "Karilynn",
    city: "Honolulu",
    routing_number: 224887802,
    state: "Hawaii",
    zip: "96850",
  },
  {
    bank_name: "Ward",
    city: "Peoria",
    routing_number: 331421157,
    state: "Illinois",
    zip: "61656",
  },
  {
    bank_name: "Samson",
    city: "Jamaica",
    routing_number: 593655726,
    state: "New York",
    zip: "11436",
  },
  {
    bank_name: "Delmer",
    city: "Denver",
    routing_number: 423427452,
    state: "Colorado",
    zip: "80270",
  },
  {
    bank_name: "Walsh",
    city: "Oklahoma City",
    routing_number: 228880440,
    state: "Oklahoma",
    zip: "73135",
  },
  {
    bank_name: "Trula",
    city: "Bozeman",
    routing_number: 811244802,
    state: "Montana",
    zip: "59771",
  },
  {
    bank_name: "Viv",
    city: "Albuquerque",
    routing_number: 692682035,
    state: "New Mexico",
    zip: "87180",
  },
  {
    bank_name: "Dollie",
    city: "San Antonio",
    routing_number: 707249543,
    state: "Texas",
    zip: "78291",
  },
  {
    bank_name: "Bernardo",
    city: "Richmond",
    routing_number: 859971484,
    state: "Virginia",
    zip: "23293",
  },
  {
    bank_name: "Rosita",
    city: "Albany",
    routing_number: 675519439,
    state: "New York",
    zip: "12237",
  },
  {
    bank_name: "Erin",
    city: "Denver",
    routing_number: 498873111,
    state: "Colorado",
    zip: "80209",
  },
  {
    bank_name: "Melba",
    city: "Clearwater",
    routing_number: 535335163,
    state: "Florida",
    zip: "33758",
  },
  {
    bank_name: "Sandor",
    city: "New Orleans",
    routing_number: 916490031,
    state: "Louisiana",
    zip: "70183",
  },
  {
    bank_name: "Petronella",
    city: "Omaha",
    routing_number: 747464677,
    state: "Nebraska",
    zip: "68134",
  },
  {
    bank_name: "Eunice",
    city: "Lansing",
    routing_number: 407981014,
    state: "Michigan",
    zip: "48930",
  },
  {
    bank_name: "Mannie",
    city: "New York City",
    routing_number: 304952129,
    state: "New York",
    zip: "10184",
  },
  {
    bank_name: "Avery",
    city: "Tulsa",
    routing_number: 936501142,
    state: "Oklahoma",
    zip: "74126",
  },
  {
    bank_name: "Valene",
    city: "Fairbanks",
    routing_number: 173060444,
    state: "Alaska",
    zip: "99790",
  },
  {
    bank_name: "Janifer",
    city: "Lubbock",
    routing_number: 530484329,
    state: "Texas",
    zip: "79491",
  },
  {
    bank_name: "Griffin",
    city: "Sacramento",
    routing_number: 288115184,
    state: "California",
    zip: "94297",
  },
  {
    bank_name: "Lilly",
    city: "Kissimmee",
    routing_number: 842286875,
    state: "Florida",
    zip: "34745",
  },
  {
    bank_name: "Yoko",
    city: "Washington",
    routing_number: 184481379,
    state: "District of Columbia",
    zip: "20337",
  },
  {
    bank_name: "Cherlyn",
    city: "North Little Rock",
    routing_number: 514038385,
    state: "Arkansas",
    zip: "72118",
  },
  {
    bank_name: "Pen",
    city: "Los Angeles",
    routing_number: 937516362,
    state: "California",
    zip: "90101",
  },
  {
    bank_name: "Court",
    city: "Washington",
    routing_number: 426945562,
    state: "District of Columbia",
    zip: "20503",
  },
  {
    bank_name: "Marcy",
    city: "Norfolk",
    routing_number: 154955464,
    state: "Virginia",
    zip: "23520",
  },
  {
    bank_name: "Kizzie",
    city: "Austin",
    routing_number: 588297101,
    state: "Texas",
    zip: "78732",
  },
  {
    bank_name: "Marlane",
    city: "Peoria",
    routing_number: 312666186,
    state: "Illinois",
    zip: "61651",
  },
  {
    bank_name: "Parker",
    city: "Jackson",
    routing_number: 473229698,
    state: "Mississippi",
    zip: "39282",
  },
  {
    bank_name: "Kylila",
    city: "Birmingham",
    routing_number: 310231910,
    state: "Alabama",
    zip: "35295",
  },
  {
    bank_name: "Harmony",
    city: "Charlotte",
    routing_number: 270117236,
    state: "North Carolina",
    zip: "28210",
  },
  {
    bank_name: "Dehlia",
    city: "Lansing",
    routing_number: 713767044,
    state: "Michigan",
    zip: "48901",
  },
  {
    bank_name: "Andres",
    city: "Flushing",
    routing_number: 836255617,
    state: "New York",
    zip: "11388",
  },
  {
    bank_name: "Stevie",
    city: "Lincoln",
    routing_number: 641717493,
    state: "Nebraska",
    zip: "68583",
  },
  {
    bank_name: "Alicia",
    city: "Springfield",
    routing_number: 812862008,
    state: "Missouri",
    zip: "65810",
  },
  {
    bank_name: "Sharia",
    city: "Abilene",
    routing_number: 807061088,
    state: "Texas",
    zip: "79605",
  },
  {
    bank_name: "Rocky",
    city: "Los Angeles",
    routing_number: 366140410,
    state: "California",
    zip: "90005",
  },
];
