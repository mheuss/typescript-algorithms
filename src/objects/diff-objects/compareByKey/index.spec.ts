import { cloneDeep } from "lodash";
import { newFile, originalFile } from "../mockData";
import { IBankData } from "../mockTypes";
import { CompareByKey } from "./index";

describe("compareByKey unit tests", () => {
  describe("Hash map creation test", () => {
    it("Should correctly map the files", () => {
      const file = new CompareByKey<IBankData>(
        originalFile,
        newFile,
        "routing_number"
      );
      const ORIGINAL_FILE = "originalObject";
      const NEW_FILE = "newObject";

      expect(file[ORIGINAL_FILE]).toMatchSnapshot();
      expect(file[NEW_FILE]).toMatchSnapshot();
    });

    it("Should ensure we are protected from an undefined key", () => {
      const modifiedOriginalFile = cloneDeep(originalFile);

      delete modifiedOriginalFile[0].routing_number;
      try {
        const file = new CompareByKey<IBankData>(
          modifiedOriginalFile,
          newFile,
          "routing_number"
        );
        fail(`Should have thrown ${file}`);
      } catch (e) {
        expect(e).toMatchSnapshot();
      }
    });

    it("Should ensure we are protected from a duplicate key", () => {
      const modifiedOriginalFile = cloneDeep(originalFile);

      modifiedOriginalFile[1].routing_number =
        modifiedOriginalFile[0].routing_number;
      try {
        const file = new CompareByKey<IBankData>(
          modifiedOriginalFile,
          newFile,
          "routing_number"
        );
        fail(`Should have thrown ${file}`);
      } catch (e) {
        expect(e).toMatchSnapshot();
      }
    });

    it("Should give me a list of data no longer in file", () => {
      const file = new CompareByKey<IBankData>(
        originalFile,
        newFile,
        "routing_number"
      );
      expect(file.getDataNoLongerInFile()).toMatchSnapshot();
    });

    it("Should give me a list of new data in file", () => {
      const file = new CompareByKey<IBankData>(
        originalFile,
        newFile,
        "routing_number"
      );
      expect(file.getNewData()).toMatchSnapshot();
    });
  });
});
