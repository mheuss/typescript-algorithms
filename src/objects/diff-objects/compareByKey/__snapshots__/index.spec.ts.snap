// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compareByKey unit tests Hash map creation test Should correctly map the files 1`] = `
{
  "145543598": {
    "bank_name": "Phillipe",
    "city": "Largo",
    "routing_number": 145543598,
    "state": "Florida",
    "zip": "34643",
  },
  "149772497": {
    "bank_name": "Shelby",
    "city": "Shreveport",
    "routing_number": 149772497,
    "state": "Louisiana",
    "zip": "71130",
  },
  "151532905": {
    "bank_name": "Danielle",
    "city": "Midland",
    "routing_number": 151532905,
    "state": "Texas",
    "zip": "78703",
  },
  "154955464": {
    "bank_name": "Marcy",
    "city": "Norfolk",
    "routing_number": 154955464,
    "state": "Virginia",
    "zip": "23520",
  },
  "173060444": {
    "bank_name": "Valene",
    "city": "Fairbanks",
    "routing_number": 173060444,
    "state": "Alaska",
    "zip": "99790",
  },
  "184481379": {
    "bank_name": "Yoko",
    "city": "Washington",
    "routing_number": 184481379,
    "state": "District of Columbia",
    "zip": "20337",
  },
  "202463187": {
    "bank_name": "Francis",
    "city": "Worcester",
    "routing_number": 202463187,
    "state": "Massachusetts",
    "zip": "01605",
  },
  "224887802": {
    "bank_name": "Karilynn",
    "city": "Honolulu",
    "routing_number": 224887802,
    "state": "Hawaii",
    "zip": "96850",
  },
  "228880440": {
    "bank_name": "Walsh",
    "city": "Oklahoma City",
    "routing_number": 228880440,
    "state": "Oklahoma",
    "zip": "73135",
  },
  "270117236": {
    "bank_name": "Harmony",
    "city": "Charlotte",
    "routing_number": 270117236,
    "state": "North Carolina",
    "zip": "28210",
  },
  "279027748": {
    "bank_name": "Kirsteni",
    "city": "Seattle",
    "routing_number": 279027748,
    "state": "Washington",
    "zip": "98104",
  },
  "288115184": {
    "bank_name": "Griffin",
    "city": "Sacramento",
    "routing_number": 288115184,
    "state": "California",
    "zip": "94297",
  },
  "304952129": {
    "bank_name": "Mannie",
    "city": "New York City",
    "routing_number": 304952129,
    "state": "New York",
    "zip": "10184",
  },
  "310231910": {
    "bank_name": "Kylila",
    "city": "Birmingham",
    "routing_number": 310231910,
    "state": "Alabama",
    "zip": "35295",
  },
  "312666186": {
    "bank_name": "Marlane",
    "city": "Peoria",
    "routing_number": 312666186,
    "state": "Illinois",
    "zip": "61651",
  },
  "316489490": {
    "bank_name": "Tait",
    "city": "Saint Louis",
    "routing_number": 316489490,
    "state": "Missouri",
    "zip": "63126",
  },
  "331421157": {
    "bank_name": "Ward",
    "city": "Chicago",
    "routing_number": 331421157,
    "state": "Illinois",
    "zip": "61656",
  },
  "407981014": {
    "bank_name": "Eunice",
    "city": "Lansing",
    "routing_number": 407981014,
    "state": "Michigan",
    "zip": "48930",
  },
  "423427452": {
    "bank_name": "Delmer",
    "city": "Denver",
    "routing_number": 423427452,
    "state": "Colorado",
    "zip": "80270",
  },
  "426945562": {
    "bank_name": "Court",
    "city": "Washington",
    "routing_number": 426945562,
    "state": "District of Columbia",
    "zip": "20503",
  },
  "473229698": {
    "bank_name": "Parker",
    "city": "Jackson",
    "routing_number": 473229698,
    "state": "Mississippi",
    "zip": "39282",
  },
  "498873111": {
    "bank_name": "Erin",
    "city": "Denver",
    "routing_number": 498873111,
    "state": "Colorado",
    "zip": "80209",
  },
  "514038385": {
    "bank_name": "Cherlyn",
    "city": "North Little Rock",
    "routing_number": 514038385,
    "state": "Arkansas",
    "zip": "72118",
  },
  "527362918": {
    "bank_name": "Bili",
    "city": "Saint Paul",
    "routing_number": 527362918,
    "state": "Minnesota",
    "zip": "55146",
  },
  "530484329": {
    "bank_name": "Janifer",
    "city": "Lubbock",
    "routing_number": 530484329,
    "state": "Texas",
    "zip": "79491",
  },
  "535335163": {
    "bank_name": "Melba",
    "city": "Clearwater",
    "routing_number": 535335163,
    "state": "Florida",
    "zip": "33758",
  },
  "547444708": {
    "bank_name": "Dorelia",
    "city": "Memphis",
    "routing_number": 547444708,
    "state": "Tennessee",
    "zip": "38131",
  },
  "588297101": {
    "bank_name": "Kizzie",
    "city": "Austin",
    "routing_number": 588297101,
    "state": "Texas",
    "zip": "78732",
  },
  "593655726": {
    "bank_name": "Samson",
    "city": "Jamaica",
    "routing_number": 593655726,
    "state": "New York",
    "zip": "11436",
  },
  "617706482": {
    "bank_name": "Carolin",
    "city": "Saint Paul",
    "routing_number": 617706482,
    "state": "Minnesota",
    "zip": "55103",
  },
  "675519439": {
    "bank_name": "Rosita",
    "city": "Albany",
    "routing_number": 675519439,
    "state": "New York",
    "zip": "12237",
  },
  "692682035": {
    "bank_name": "Viv",
    "city": "Albuquerque",
    "routing_number": 692682035,
    "state": "New Mexico",
    "zip": "87180",
  },
  "701886244": {
    "bank_name": "Gwenni",
    "city": "Wichita",
    "routing_number": 701886244,
    "state": "Kansas",
    "zip": "67215",
  },
  "707249543": {
    "bank_name": "Dollie",
    "city": "San Antonio",
    "routing_number": 707249543,
    "state": "Texas",
    "zip": "78291",
  },
  "713767044": {
    "bank_name": "Dehlia",
    "city": "Lansing",
    "routing_number": 713767044,
    "state": "Michigan",
    "zip": "48901",
  },
  "732919062": {
    "bank_name": "Merle",
    "city": "Denver",
    "routing_number": 732919062,
    "state": "Colorado",
    "zip": "80930",
  },
  "747464677": {
    "bank_name": "Petronella",
    "city": "Omaha",
    "routing_number": 747464677,
    "state": "Nebraska",
    "zip": "68134",
  },
  "760086826": {
    "bank_name": "Jaime",
    "city": "Indianapolis",
    "routing_number": 760086826,
    "state": "Indiana",
    "zip": "46295",
  },
  "811244802": {
    "bank_name": "Trula",
    "city": "Bozeman",
    "routing_number": 811244802,
    "state": "Montana",
    "zip": "59771",
  },
  "842286875": {
    "bank_name": "Lilly",
    "city": "Kissimmee",
    "routing_number": 842286875,
    "state": "Florida",
    "zip": "34745",
  },
  "859971484": {
    "bank_name": "Bernardo",
    "city": "Richmond",
    "routing_number": 859971484,
    "state": "Virginia",
    "zip": "23293",
  },
  "916490031": {
    "bank_name": "Sandor",
    "city": "New Orleans",
    "routing_number": 916490031,
    "state": "Louisiana",
    "zip": "70183",
  },
  "920233367": {
    "bank_name": "Jolynn",
    "city": "Houston",
    "routing_number": 920233367,
    "state": "Texas",
    "zip": "77010",
  },
  "936501142": {
    "bank_name": "Avery",
    "city": "Tulsa",
    "routing_number": 936501142,
    "state": "Oklahoma",
    "zip": "74126",
  },
  "937516362": {
    "bank_name": "Pen",
    "city": "Los Angeles",
    "routing_number": 937516362,
    "state": "California",
    "zip": "90101",
  },
}
`;

exports[`compareByKey unit tests Hash map creation test Should correctly map the files 2`] = `
{
  "149772497": {
    "bank_name": "Shelby",
    "city": "Shreveport",
    "routing_number": 149772497,
    "state": "Louisiana",
    "zip": "71130",
  },
  "151532905": {
    "bank_name": "Danielle",
    "city": "Austin",
    "routing_number": 151532905,
    "state": "Texas",
    "zip": "78703",
  },
  "154955464": {
    "bank_name": "Marcy",
    "city": "Norfolk",
    "routing_number": 154955464,
    "state": "Virginia",
    "zip": "23520",
  },
  "173060444": {
    "bank_name": "Valene",
    "city": "Fairbanks",
    "routing_number": 173060444,
    "state": "Alaska",
    "zip": "99790",
  },
  "184481379": {
    "bank_name": "Yoko",
    "city": "Washington",
    "routing_number": 184481379,
    "state": "District of Columbia",
    "zip": "20337",
  },
  "202463187": {
    "bank_name": "Francis",
    "city": "Worcester",
    "routing_number": 202463187,
    "state": "Massachusetts",
    "zip": "01605",
  },
  "224887802": {
    "bank_name": "Karilynn",
    "city": "Honolulu",
    "routing_number": 224887802,
    "state": "Hawaii",
    "zip": "96850",
  },
  "228880440": {
    "bank_name": "Walsh",
    "city": "Oklahoma City",
    "routing_number": 228880440,
    "state": "Oklahoma",
    "zip": "73135",
  },
  "270117236": {
    "bank_name": "Harmony",
    "city": "Charlotte",
    "routing_number": 270117236,
    "state": "North Carolina",
    "zip": "28210",
  },
  "279027748": {
    "bank_name": "Kirsteni",
    "city": "Seattle",
    "routing_number": 279027748,
    "state": "Washington",
    "zip": "98104",
  },
  "288115184": {
    "bank_name": "Griffin",
    "city": "Sacramento",
    "routing_number": 288115184,
    "state": "California",
    "zip": "94297",
  },
  "304952129": {
    "bank_name": "Mannie",
    "city": "New York City",
    "routing_number": 304952129,
    "state": "New York",
    "zip": "10184",
  },
  "310231910": {
    "bank_name": "Kylila",
    "city": "Birmingham",
    "routing_number": 310231910,
    "state": "Alabama",
    "zip": "35295",
  },
  "312666186": {
    "bank_name": "Marlane",
    "city": "Peoria",
    "routing_number": 312666186,
    "state": "Illinois",
    "zip": "61651",
  },
  "316489490": {
    "bank_name": "Tait",
    "city": "Saint Louis",
    "routing_number": 316489490,
    "state": "Missouri",
    "zip": "63126",
  },
  "331421157": {
    "bank_name": "Ward",
    "city": "Peoria",
    "routing_number": 331421157,
    "state": "Illinois",
    "zip": "61656",
  },
  "366140410": {
    "bank_name": "Rocky",
    "city": "Los Angeles",
    "routing_number": 366140410,
    "state": "California",
    "zip": "90005",
  },
  "407981014": {
    "bank_name": "Eunice",
    "city": "Lansing",
    "routing_number": 407981014,
    "state": "Michigan",
    "zip": "48930",
  },
  "423427452": {
    "bank_name": "Delmer",
    "city": "Denver",
    "routing_number": 423427452,
    "state": "Colorado",
    "zip": "80270",
  },
  "426945562": {
    "bank_name": "Court",
    "city": "Washington",
    "routing_number": 426945562,
    "state": "District of Columbia",
    "zip": "20503",
  },
  "473229698": {
    "bank_name": "Parker",
    "city": "Jackson",
    "routing_number": 473229698,
    "state": "Mississippi",
    "zip": "39282",
  },
  "498873111": {
    "bank_name": "Erin",
    "city": "Denver",
    "routing_number": 498873111,
    "state": "Colorado",
    "zip": "80209",
  },
  "514038385": {
    "bank_name": "Cherlyn",
    "city": "North Little Rock",
    "routing_number": 514038385,
    "state": "Arkansas",
    "zip": "72118",
  },
  "527362918": {
    "bank_name": "Bili",
    "city": "Saint Paul",
    "routing_number": 527362918,
    "state": "Minnesota",
    "zip": "55146",
  },
  "530484329": {
    "bank_name": "Janifer",
    "city": "Lubbock",
    "routing_number": 530484329,
    "state": "Texas",
    "zip": "79491",
  },
  "535335163": {
    "bank_name": "Melba",
    "city": "Clearwater",
    "routing_number": 535335163,
    "state": "Florida",
    "zip": "33758",
  },
  "588297101": {
    "bank_name": "Kizzie",
    "city": "Austin",
    "routing_number": 588297101,
    "state": "Texas",
    "zip": "78732",
  },
  "593655726": {
    "bank_name": "Samson",
    "city": "Jamaica",
    "routing_number": 593655726,
    "state": "New York",
    "zip": "11436",
  },
  "617706482": {
    "bank_name": "Carolin",
    "city": "Saint Paul",
    "routing_number": 617706482,
    "state": "Minnesota",
    "zip": "55103",
  },
  "641717493": {
    "bank_name": "Stevie",
    "city": "Lincoln",
    "routing_number": 641717493,
    "state": "Nebraska",
    "zip": "68583",
  },
  "675519439": {
    "bank_name": "Rosita",
    "city": "Albany",
    "routing_number": 675519439,
    "state": "New York",
    "zip": "12237",
  },
  "692682035": {
    "bank_name": "Viv",
    "city": "Albuquerque",
    "routing_number": 692682035,
    "state": "New Mexico",
    "zip": "87180",
  },
  "707249543": {
    "bank_name": "Dollie",
    "city": "San Antonio",
    "routing_number": 707249543,
    "state": "Texas",
    "zip": "78291",
  },
  "713767044": {
    "bank_name": "Dehlia",
    "city": "Lansing",
    "routing_number": 713767044,
    "state": "Michigan",
    "zip": "48901",
  },
  "732919062": {
    "bank_name": "Merle",
    "city": "Colorado Springs",
    "routing_number": 732919062,
    "state": "Colorado",
    "zip": "80930",
  },
  "747464677": {
    "bank_name": "Petronella",
    "city": "Omaha",
    "routing_number": 747464677,
    "state": "Nebraska",
    "zip": "68134",
  },
  "807061088": {
    "bank_name": "Sharia",
    "city": "Abilene",
    "routing_number": 807061088,
    "state": "Texas",
    "zip": "79605",
  },
  "811244802": {
    "bank_name": "Trula",
    "city": "Bozeman",
    "routing_number": 811244802,
    "state": "Montana",
    "zip": "59771",
  },
  "812862008": {
    "bank_name": "Alicia",
    "city": "Springfield",
    "routing_number": 812862008,
    "state": "Missouri",
    "zip": "65810",
  },
  "836255617": {
    "bank_name": "Andres",
    "city": "Flushing",
    "routing_number": 836255617,
    "state": "New York",
    "zip": "11388",
  },
  "842286875": {
    "bank_name": "Lilly",
    "city": "Kissimmee",
    "routing_number": 842286875,
    "state": "Florida",
    "zip": "34745",
  },
  "859971484": {
    "bank_name": "Bernardo",
    "city": "Richmond",
    "routing_number": 859971484,
    "state": "Virginia",
    "zip": "23293",
  },
  "916490031": {
    "bank_name": "Sandor",
    "city": "New Orleans",
    "routing_number": 916490031,
    "state": "Louisiana",
    "zip": "70183",
  },
  "936501142": {
    "bank_name": "Avery",
    "city": "Tulsa",
    "routing_number": 936501142,
    "state": "Oklahoma",
    "zip": "74126",
  },
  "937516362": {
    "bank_name": "Pen",
    "city": "Los Angeles",
    "routing_number": 937516362,
    "state": "California",
    "zip": "90101",
  },
}
`;

exports[`compareByKey unit tests Hash map creation test Should ensure we are protected from a duplicate key 1`] = `
{
  "message": "Key is not unique ",
}
`;

exports[`compareByKey unit tests Hash map creation test Should ensure we are protected from an undefined key 1`] = `
{
  "message": "You cannot have an undefined key name",
}
`;

exports[`compareByKey unit tests Hash map creation test Should give me a list of data no longer in file 1`] = `
[
  {
    "bank_name": "Phillipe",
    "city": "Largo",
    "routing_number": 145543598,
    "state": "Florida",
    "zip": "34643",
  },
  {
    "bank_name": "Dorelia",
    "city": "Memphis",
    "routing_number": 547444708,
    "state": "Tennessee",
    "zip": "38131",
  },
  {
    "bank_name": "Gwenni",
    "city": "Wichita",
    "routing_number": 701886244,
    "state": "Kansas",
    "zip": "67215",
  },
  {
    "bank_name": "Jaime",
    "city": "Indianapolis",
    "routing_number": 760086826,
    "state": "Indiana",
    "zip": "46295",
  },
  {
    "bank_name": "Jolynn",
    "city": "Houston",
    "routing_number": 920233367,
    "state": "Texas",
    "zip": "77010",
  },
]
`;

exports[`compareByKey unit tests Hash map creation test Should give me a list of new data in file 1`] = `
[
  {
    "bank_name": "Rocky",
    "city": "Los Angeles",
    "routing_number": 366140410,
    "state": "California",
    "zip": "90005",
  },
  {
    "bank_name": "Stevie",
    "city": "Lincoln",
    "routing_number": 641717493,
    "state": "Nebraska",
    "zip": "68583",
  },
  {
    "bank_name": "Sharia",
    "city": "Abilene",
    "routing_number": 807061088,
    "state": "Texas",
    "zip": "79605",
  },
  {
    "bank_name": "Alicia",
    "city": "Springfield",
    "routing_number": 812862008,
    "state": "Missouri",
    "zip": "65810",
  },
  {
    "bank_name": "Andres",
    "city": "Flushing",
    "routing_number": 836255617,
    "state": "New York",
    "zip": "11388",
  },
]
`;
